src/main/java/com/teragrep/aer_02/EventDataConsumer.java

```diff
@@ -0,0 +1,203 @@
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+final class EventDataConsumer implements AutoCloseable {
+
+    // Note: Checkpointing is handled automatically.
+    private final Output output;
+    private final String realHostName;
+    private final SyslogConfig syslogConfig;
+    private final MetricRegistry metricRegistry;
+    private final JmxReporter jmxReporter;
+    private final Slf4jReporter slf4jReporter;
+
+    EventDataConsumer(Sourceable configSource, int prometheusPort) {
+        this(configSource, new MetricRegistry(), prometheusPort);
+    }
+
+    EventDataConsumer(Sourceable configSource, MetricRegistry metricRegistry, int prometheusPort) {
+        this(
+                configSource,
+                new DefaultOutput("defaultOutput", new RelpConfig(configSource), metricRegistry),
+                metricRegistry,
+                prometheusPort
+        );
+    }
+
+    EventDataConsumer(Sourceable configSource, Output output, MetricRegistry metricRegistry, int prometheusPort) {
+        this.metricRegistry = metricRegistry;
+        this.output = output;
+        this.realHostName = getRealHostName();
+        this.syslogConfig = new SyslogConfig(configSource);
+
+        this.jmxReporter = JmxReporter.forRegistry(metricRegistry).build();
+        this.slf4jReporter = Slf4jReporter
+                .forRegistry(metricRegistry)
+                .outputTo(LoggerFactory.getLogger(EventDataConsumer.class))
+                .convertRatesTo(TimeUnit.SECONDS)
+                .convertDurationsTo(TimeUnit.MILLISECONDS)
+                .build();
+        startMetrics();
+    }
+
+    private String getRealHostName() {
+        String hostname;
+        try {
+            hostname = InetAddress.getLocalHost().getHostName();
+        }
+        catch (UnknownHostException e) {
+            hostname = "localhost";
+        }
+        return hostname;
+    }
+
+    private void startMetrics() {
+        this.jmxReporter.start();
+        this.slf4jReporter.start(1, TimeUnit.MINUTES);
+
+        // prometheus-exporter
+        CollectorRegistry.defaultRegistry.register(new DropwizardExports(metricRegistry));
+    }
+
+    public void accept(
+            String eventData,
+            Map<String, Object> partitionContext,
+            ZonedDateTime enqueuedTime,
+            String offset,
+            Map<String, Object> props,
+            Map<String, Object> systemProps
+    ) {
+
+        String partitionId = String.valueOf(partitionContext.get("PartitionId"));
+        metricRegistry.gauge(name(EventDataConsumer.class, "latency-seconds", partitionId), () -> new Gauge<Long>() {
+
+            @Override
+            public Long getValue() {
+                return Instant.now().getEpochSecond() - enqueuedTime.toInstant().getEpochSecond();
+            }
+        });
```

Comment:
A lambda can be used here, but it makes the code a little less clear

```suggestion
        metricRegistry.gauge(name(EventDataConsumer.class, "latency-seconds", partitionId), () -> (Gauge<Long>) () ->
            Instant.now().getEpochSecond() - enqueuedTime.toInstant().getEpochSecond()
        );
```